lib/scanner/walk.go

```diff
@@ -707,5 +708,8 @@ func CreateFileInfo(fi fs.FileInfo, name string, filesystem fs.Filesystem, scanO
 	} else {
 		f.InodeChangeNs = 0
 	}
+	if recvEnc {
+		f.Size -= protocol.EncryptionTrailerSize(f)
```

Comment:
This is wrong, because it's the size of the encryption trailer if we would write the current fileinfo to it, but that's not the fileinfo that is actually written to the file (or we wouldn't have to write it to begin with). We need to instead read the size from the file in question...